### ISSUE TITLE
How to integrate flutter engine to flutter sdk?

### ISSUE THREAD

**Issue by II11II** [flutter/flutter#134426]
I have compiled my own custom engine successfully according to the docs.
Currently, I indicate the path of the compiled host and client engine files in order to use the custom engine.
``` flutter run --local-engine=client_xxx --local-engine-host=host_xxx```

However, I would like to distribute my custom engine with (custom) [flutter sdk](https://github.com/flutter/flutter) to others without making them add extra commands while they are building their apps.

P.s. I read the docs but could not find how to compile flutter SDK itself with a custom engine.

How can I achieve this behaviour?

**Comment by darshankawar** [comment]
@II11II 
Have you already look at https://github.com/flutter/flutter/wiki/Compiling-the-engine to see if it helps ?

**Comment by II11II** [comment]
Hi @darshankawar 
Yes, I have already taken a look at https://github.com/flutter/flutter/wiki/Compiling-the-engine
And thanks to this link I could compile a custom engine.

But It seems you misunderstood the question

I cannot find the documentation where written how to replace a custom engine in Flutter SDK itself without adding custom parameters  ` flutter run --local-engine=client_xxx --local-engine-host=host_xxx`.

So in other words, the end user will download custom flutter SDK(which uses my custom engine by default) from my own git repository and use it to create apps for platforms

**Comment by chinmaygarde** [comment]
> I cannot find the documentation where written how to replace a custom engine in Flutter SDK itself.

The tool doesn't support that. Compiling local engines is a local development workflow. You will have to setup a custom CI workflow to distribute locally built engines.

**Comment by II11II** [comment]
> Compiling local engines is a local development workflow. 

It is obvious

> You will have to set up a custom CI workflow to distribute locally built engines.

Is there any official doc, on how it is done on the Flutter repository?
